packages/docusaurus-theme-search-algolia/src/theme/hooks/useSearchQuery.js

```diff
@@ -7,12 +7,14 @@
 
 import {useHistory, useLocation} from '@docusaurus/router';
 import ExecutionEnvironment from '@docusaurus/ExecutionEnvironment';
+import useBaseUrl from '@docusaurus/useBaseUrl';
 
 const SEARCH_PARAM_QUERY = 'q';
 
 function useSearchQuery() {
   const history = useHistory();
   const location = useLocation();
+  const baseUrl = useBaseUrl('');
```

Comment:
You should use it as such - 

```js
const newPath = useBaseUrl(`search?q=${searchValue}`);
```

Reply:
> Don’t call Hooks inside loops, conditions, or **nested functions**

Would that work? I do not think so. Therefore `siteConfig.baseUrl` is a good option.

Reply:
This is a hook, why would it not work?

Reply:
Try it yourself and you will see that it does not work.

```js
    navigateToSearchPage: (searchValue) => { 
      const searchUrl = useBaseUrl(`search?q=${searchValue}`); // Not work - call hook in nested functions.
      history.push(searchUrl);
    },
```

Reply:
If we put it at the top level of the function scope, alongside `useLocation()` and `useHistory()`, and not in the `navigateToSearchPage()` function, would that not work? That is my suggestion.

Reply:
Nope, this way also will not work.

Reply:
> Try it yourself and you will see that it does not work.
> 
> ```js
>     navigateToSearchPage: (searchValue) => { 
>       const searchUrl = useBaseUrl(`search?q=${searchValue}`); // Not work - call hook in nested functions.
>       history.push(searchUrl);
>     },
> ```
I confirmed this did not work because of the error "Invalid hook call".

Reply:
> If we put it at the top level of the function scope, alongside `useLocation()` and `useHistory()`, and not in the `navigateToSearchPage()` function, would that not work? That is my suggestion.

Thanks, but in the top level of the function scope, we cannot get `searchValue` defined in `navigateToSearchPage()`. So your suggestion would not work, I think.

Reply:
You're right, I failed to realize we need `searchValue`. Hmm I guess this is fine then. Mind leaving a comment in the code and why we're doing this? Could add a link to this PR :)

Reply:
@yangshun @lex111 Thanks for your reviews, guys.
I will add the comment including this PR URL.